assets/prometheus-k8s/rules.yaml

```diff
@@ -362,6 +362,14 @@ spec:
     rules:
     - expr: sum(openshift_build_total{job="kubernetes-apiservers",phase="Error"})/(sum(openshift_build_total{job="kubernetes-apiservers",phase=~"Failed|Complete|Error"}))
       record: build_error_rate
+  - name: openshift-sre.rules
+    rules:
+    - expr: sum(apiserver_request_count{job="apiserver"}) by (code)
```

Comment:
The rule is first rate then sum, as otherwise counter resets cannot be accounted for correctly. We need to sum the rate of requests here, as in:

```
sum(rate(apiserver_request_count{job="apiserver"}[5m])) by (code)
```

And for what it's worth, we already have an [alerting rule for this](https://github.com/openshift/cluster-monitoring-operator/blob/1b2c8054cbe37c686b50301aee2ba05bad8d2c8b/assets/prometheus-k8s/rules.yaml#L780-L799), and we're already reporting all firing alerting rules through telemetry.

Reply:
updated the metric to sum(rate(apiserver_request_count{job="apiserver"}[5m])) by (code)